src/Form/SearchSolrForm.php

```diff
@@ -0,0 +1,234 @@
+        ],
+      ],
+    ];
+
+    $form['auth']['username'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Username'),
+      '#default_value' => $solr_server_config->get('backend_config.connector_config.username'),
+    ];
+
+    $form['auth']['password'] = [
+      '#type' => 'password',
+      '#title' => $this->t('Password'),
+      '#description' => $this->t('If this field is left blank and the HTTP username is filled out, the current password will not be changed.'),
+    ];
+
+    $form['actions'] = [
+      'continue' => [
+        '#type' => 'submit',
+        '#value' => $this->t('Continue'),
+      ],
+      '#type' => 'actions',
+    ];
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validateForm(array &$form, FormStateInterface $form_state) {
+    $values = $form_state->getValues();
+    if (isset($values['port']) && (!is_numeric($values['port']) || $values['port'] < 0 || $values['port'] > 65535)) {
+      $form_state->setError($form['connector_set']['port'], $this->t('The port has to be an integer between 0 and 65535.'));
+    }
+    if (!empty($values['path']) && strpos($values['path'], '/') !== 0) {
+      $form_state->setError($form['connector_set']['path'], $this->t('If provided the path has to start with "/".'));
+    }
+    if (!empty($values['core']) && strpos($values['core'], '/') === 0) {
+      $form_state->setError($form['connector_set']['core'], $this->t('The core must not start with "/".'));
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $values = $form_state->getValues();
+    // Save solr server configuration.
+    $solr_server_config = $this->configFactory->getEditable('search_api.server.solr_search');
+    $solr_server_config->set('backend_config.connector', $values['connector']);
+    $solr_server_config->set('backend_config.connector_config.scheme', $values['scheme']);
+    $solr_server_config->set('backend_config.connector_config.host', $values['host']);
+    $solr_server_config->set('backend_config.connector_config.port', $values['port']);
+    $solr_server_config->set('backend_config.connector_config.path', $values['path']);
+    $solr_server_config->set('backend_config.connector_config.core', $values['core']);
+    $solr_server_config->set('backend_config.connector_config.timeout', $values['timeout']);
+    $solr_server_config->set('backend_config.connector_config.index_timeout', $values['index_timeout']);
+    $solr_server_config->set('backend_config.connector_config.optimize_timeout', $values['optimize_timeout']);
+    $solr_server_config->set('backend_config.connector_config.commit_within', $values['commit_within']);
+    // For basic_auth also save username and password.
+    if ('basic_auth' == $values['connector']) {
+      // Ignore empty password submissions if the user didn't change it.
+      if (($values['password'] === '')
```

Comment:
This condition looks ugly, please find a way to simplify it

Reply:
Fixed